Content.Server/Traits/Assorted/HeirloomSystem.cs

```diff
@@ -0,0 +1,39 @@
+using System.Linq;
+using Content.Shared.Mood;
+using Content.Shared.Traits.Assorted.Components;
+
+namespace Content.Server.Traits.Assorted;
+
+public sealed class HeirloomSystem : EntitySystem
+{
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        //TODO: This could probably use some optimization
+        var query = EntityManager.EntityQueryEnumerator<HeirloomHaverComponent>();
+        while (query.MoveNext(out var uid, out var comp))
+        {
+            var children = RecursiveGetAllChildren(uid);
```

Comment:
This is an fairly expensive function. HeirloomHavers really need to have a 'timer' to make sure it's not doing this on every frame. As you've mentioned earlier, it can be done once every few minutes. So that people don't immediately notice that their heirloom is gone. But also for performance.